scipy/stats/tests/test_morestats.py

```diff
@@ -1118,6 +1118,28 @@ def test_auto(self):
         d = np.arange(1, 27)
         assert_equal(stats.wilcoxon(d), stats.wilcoxon(d, mode="approx"))
 
+    def test_nan_policy(self):
+
+        x1 = np.array([1, 2, 3])
+        x2 = np.array([1, 2, 3, np.nan])
+        assert_equal(stats.wilcoxon(x1), stats.wilcoxon(x2, nan_policy="omit"))
+        assert_equal((np.nan, np.nan),
+                     stats.wilcoxon(x2, nan_policy="propagate"))
+
+        assert_raises(ValueError, stats.wilcoxon, x2, nan_policy='raise')
+
+        x3 = np.array([5, 5, 5])
+        x4 = np.array([5, 5, 5, np.nan])
+
+        assert_equal(stats.wilcoxon(x1, x3),
+                     stats.wilcoxon(x1, x4, nan_policy="omit"))
+        assert_equal((np.nan, np.nan),
+                     stats.wilcoxon(x1, x4, nan_policy="propagate"))
+
+        x5 = np.array([6, 6, 6])
+        x6 = np.array([7, 7, np.nan])
+        assert_raises(ValueError, stats.wilcoxon, x5, x6, nan_policy='omit')
+
```

Comment:
Let's add one more test of `raise`. Even though it tests the same code, it looks different on the outside.

```suggestion
        with assert_raises(ValueError, match="The input contains nan"):
            stats.wilcoxon(x1, x2, nan_policy='raise')
```